worlds/witness/__init__.py

```diff
@@ -172,6 +165,63 @@ def pre_fill(self):
             first_check.place_locked_item(item)
             self.multiworld.itempool.remove(item)
 
+        itempool_difference = len(self.multiworld.get_unfilled_locations(self.player)) - len(self.multiworld.itempool)
+
+        # fill rest of item pool with junk if there is room
+        if itempool_difference > 0:
+            for i in range(0, itempool_difference):
+                self.multiworld.itempool.append(self.create_item(self.get_filler_item_name()))
+
+        # remove junk, Functioning Brain, useful items (non-door), useful door items in that order until there is room
+        if itempool_difference < 0:
+            junk = [
+                item for item in self.multiworld.itempool
+                if item.classification in {ItemClassification.filler, ItemClassification.trap}
+                and item.name != "Functioning Brain"
+            ]
+
+            f_brain = [item for item in self.multiworld.itempool if item.name == "Functioning Brain"]
+
+            usefuls = [
+                item for item in self.multiworld.itempool
+                if item.classification == ItemClassification.useful
+                and item.name not in StaticWitnessLogic.ALL_DOOR_ITEMS_AS_DICT
+            ]
+
+            removable_doors = [
+                item for item in self.multiworld.itempool
+                if item.classification == ItemClassification.useful
+                and item.name in StaticWitnessLogic.ALL_DOOR_ITEMS_AS_DICT
+            ]
+
+            self.multiworld.per_slot_randoms[self.player].shuffle(junk)
+            self.multiworld.per_slot_randoms[self.player].shuffle(usefuls)
+            self.multiworld.per_slot_randoms[self.player].shuffle(removable_doors)
+
+            removed_junk = False
+            removed_usefuls = False
+            removed_doors = False
+
+            for i in range(itempool_difference, 0):
+                if junk:
+                    self.multiworld.itempool.remove(junk.pop())
+                    removed_junk = True
+                if f_brain:
```

Comment:
```suggestion
                elif f_brain:
```
This feels like it should be elif

Reply:
Tru